subprocess_utils.hpp

```diff
@@ -0,0 +1,219 @@
+#pragma once
```

Comment:
Should we really have more than one file? I mean, it's great for organisation and all, but it means that's its one extra file users will need to include into their project

Reply:
No, I agree that it should be in just one file. But it's way easier to develop when its more spread out.

Reply:
hm, i suppose we can make subprocess_dev.hpp etc, and then make a commit hook so that a "compiled" header is made?